product_subscription/models/subscription_request.py

```diff
@@ -139,10 +139,11 @@ def validate_request(self):
         for request in self:
             partner = request.sponsor
             invoice = request.create_invoice(partner, {})
+            request.invoice = invoice
             invoice.compute_taxes()
             invoice.signal_workflow('invoice_open')
             request.send_invoice(invoice)
-            request.write({'state': 'sent', 'invoice': invoice.id})
+            request.state = 'sent'
```

Comment:
@robinkeunen you're a performing two sql query by splitting the write in two assignations

Reply:
Indeed but I need to have invoice set before `signal_workflow` and I don't want to set request to _sent_ before `send_invoice`

Reply:
ok then :)